### ISSUE TITLE
v2.6.0 fails to be processed by ngcc due to `.__ivy_ngcc_bck` files shipped to npm

### ISSUE THREAD

**Issue by gkalpak** [highcharts/highcharts-angular#215]
### Describe the bug
While upgrading from v2.4.0 to v2.6.0 on our [ngcc-validation](https://github.com/angular/ngcc-validation) repo (PR: angular/ngcc-validation#1603), we started getting the following error when compiling `highcharts-angular` with ngcc (the Angular Compatibility Compiler):

> Error: Tried to overwrite /home/circleci/repo/node_modules/highcharts-angular/lib/highcharts-chart.component.d.ts.__ivy_ngcc_bak with an ngcc back up file, which is disallowed.

It seems that v2.6.0 ships some ngcc compilation artifacts to [npm](https://unpkg.com/browse/highcharts-angular@2.6.0/) - for example the `__ivy_ngcc__` directory and `.__ivy_ngcc_bck` files. This should not happen (i.e. these artifacts are not supposed to be shipped to npm).

### Expected behavior
Expected no ngcc compilation artifacts to be shipped to npm and expected compiling the library with ngcc not to fail :grin:

### Demo
You can see a failing CI run [here](https://circleci.com/gh/angular/ngcc-validation/11659).
Basically, adding `highcharts-angular` to an Angular project and running `node_modules/.bin/ngcc --create-ivy-entry-points --first-only --properties browser module main --no-tsconfig` should be enough to reproduce the error.

### Setup used
 - NodeJS version 12
 - Angular version 10.0.5
 - TypeScript version 3.9.5

**Comment by alan-agius4** [comment]
Hi,

The issue where ngcc artefacts being present in the final distributable, might be caused because between runs the `dist` Is not being cleaned up https://github.com/highcharts/highcharts-angular/blob/24bfe42ac7b53509f38a626f31762f8a70a48d0a/highcharts-angular/ng-package.json#L4.

Also you seem to have an `ng-package.prod.json` which is not being used.

My recommendation would be to drop the production config file and also delete the `deleteDestPath` setting which is set to true by default.

Let me know if you need any assistance!

**Comment by gkalpak** [comment]
FYI, v2.7.0 seems work like a charm ([updated PR](https://github.com/angular/ngcc-validation/pull/1603) | [passing CI run](https://circleci.com/workflow-run/df312fea-11cc-4cfb-9946-1fef6799bc6f)).
Thanks for the super quick turn-around, @KacperMadej 👍

**Comment by gkalpak** [comment]
(Leaving this open in case you want to act on https://github.com/highcharts/highcharts-angular/issues/215#issuecomment-663564168, but feel free to close.)
